src/main/java/com/laytonsmith/core/MethodScriptCompiler.java

```diff
@@ -2216,6 +2216,27 @@ private static ParseTree postParseRewrite(ParseTree ast, Environment env,
 			if(newChild != null && child != newChild) {
 				ast.getChildren().set(i, newChild);
 				i--; // Allow the new child to do a rewrite step as well.
+				continue;
+			}
+
+			// In strict mode throw compile errors when encountering child statements in function arguments where
+			// statements are not acceptable because void would be an invalid argument type. This would otherwise be
+			// a runtime error in strict mode where auto-concat is not allowed and statements are used instead.
+			// This can be removed once a more comprehensive void return type check is implemented.
+			if(ast.getFileOptions().isStrict() && child.getData().val().equals(Compiler.__statements__.NAME)) {
```

Comment:
This should contain an instanceof CFunction to not match `__statements__` string literals and other terms that stringify to this.